magit-file-icons.el

```diff
@@ -55,12 +55,52 @@
   :type 'boolean
   :group 'magit-file-icons)
 
+(defcustom magit-file-icons-icon-backend 'nerd-icons
+  "Icon backend for magit-file-icons. Set this to nil
+if you want to customize `magit-file-icons-icon-for-file-func'
+or `magit-file-icons-icon-for-dir-func'."
+  :type 'symbol
+  :group 'magit-file-icons)
+
+(if (not (require 'nerd-icons nil t)) (funcall
+                                       (lambda ()
+                                        (require 'all-the-icons)
+                                        (setq magit-file-icons-icon-backend 'all-the-icons))))
+
+(defcustom magit-file-icons-icon-for-file-func nil
+  "Icon for file function. Automatically set if
+`magit-file-icons-icon-backend' is non-nil.
+Customize using (fset 'magit-file-icons-icon-for-file-func 'custom-function)."
+  :type 'symbol
+  :group 'magit-file-icons)
+
+(defcustom magit-file-icons-icon-for-dir-func nil
+  "Icon for directory function. Automatically set if
+`magit-file-icons-icon-backend' is non-nil.
+Customize using (fset 'magit-file-icons-icon-for-file-func 'custom-function)."
+  :type 'symbol
+  :group 'magit-file-icons)
+
+(defun magit-file-icons-refresh-backend ()
+  "Refresh backend according to `magit-file-icons-icon-backend'.
+Does not refresh if `magit-file-icons-icon-backend' is nil."
+  (if magit-file-icons-icon-backend
+  (if (eq magit-file-icons-icon-backend 'nerd-icons)
+      (funcall (lambda ()
+                 (fset 'magit-file-icons-icon-for-file-func 'nerd-icons-icon-for-file)
+                 (fset 'magit-file-icons-icon-for-dir-func 'nerd-icons-icon-for-dir)))
+      (funcall (lambda ()
+                 (fset 'magit-file-icons-icon-for-file-func 'all-the-icons-icon-for-file)
+                 (fset 'magit-file-icons-icon-for-dir-func 'all-the-icons-icon-for-dir))))))
+
+(magit-file-icons-refresh-backend)
```

Comment:
I think we can get rid of this line if we call the refresh function in the minor-mode anyway
```suggestion
```